### PR TITLE
WebAuthn requires either gmp or bcmatch

### PR DESCRIPTION
**Pull request by nikosdion**
### Summary of Changes

Changes the check in Joomla 4's implementation of WebAuthn to require _either_ the GMP _or_ the BCmatch PHP extension. You see, the library we use is compatible with **both**, despite what @PhilETaylor asserted :wink: 

### Testing Instructions

Disable the GMP PHP extension on your site but do enable BCmath.

Try to register a new Authenticator with WebAuthn.

### Actual result BEFORE applying this Pull Request

You get an error message that the PHP extension GMP is required to register an authenticator.

### Expected result AFTER applying this Pull Request

You can register your authentication just fine.

### Documentation Changes Required

Changed the language string `PLG_SYSTEM_WEBAUTHN_REQUIRES_GMP` to reflect reality.

### Further information

The requirement for GMP or BCmath comes from the third party library `fgrosse/PHPASN1` which is in the WebAuthn library's dependency graph.  That library had a TODO item to add a pure PHP implementation for big integer mathematics. In the past, I had used `phpseclib/bcmath_compat` to do exactly that. So I took it upon myself to file an [upstream PR](https://github.com/fgrosse/PHPASN1/pull/80) to the library.

Once the upstream PR is accepted and a new version is released we can run `composer update` on Joomla 4 and completely remove the GMP/BCmath check. You're welcome.

### CODE REVIEW

File: administrator/language/en-GB/plg_system_webauthn.ini [hunk 1]
```diff
@@ -41,5 +41,4 @@ PLG_SYSTEM_WEBAUTHN_MANAGE_HEADER_ACTIONS_LABEL="Actions"
 PLG_SYSTEM_WEBAUTHN_MANAGE_HEADER_NOMETHODS_LABEL="No authenticators have been set up yet."
 PLG_SYSTEM_WEBAUTHN_MSG_DELETED="The authenticator has been removed."
 PLG_SYSTEM_WEBAUTHN_MSG_SAVED_LABEL="The label has been saved."
-PLG_SYSTEM_WEBAUTHN_REQUIRES_GMP="The PHP Extension GMP is required to be loaded in order to add authenticators."
-
+PLG_SYSTEM_WEBAUTHN_REQUIRES_GMP="One of the PHP extensions GMP or BCmath must be loaded to add authenticators."
```
Comment (brianteeman) [inline administrator/language/en-GB/plg_system_webauthn.ini]:
```suggestion
PLG_SYSTEM_WEBAUTHN_REQUIRES_GMP="Either of the PHP extensions GMP or BCmath must be loaded to add authenticators."
```

This is a super pedantic grammar change. Use "either" when there are only two options. Use "any of" when there are more than to

Comment (nikosdion) [reply administrator/language/en-GB/plg_system_webauthn.ini]: I thought about it but "either" sounds like an "exclusive or" to me, i.e. I need to have only the GMP extension installed or only the BCmatch extension involved. I used "One of" to signal that you can have only GMP, only BCmath or both.

Given this additional context do you think I should still make the change?

Comment (richard67) [reply administrator/language/en-GB/plg_system_webauthn.ini]: Maybe "At least one of the extensions GMP or BCmath ..."?

Comment (brianteeman) [reply administrator/language/en-GB/plg_system_webauthn.ini]: @nikosdion and to me it sounds the opposite. I stick by my grammar teachers

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on b3b9765a5622a89171daeedcba4af17797d574e2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30316">issues.joomla.org/tracker/joomla-cms/30316</a>.</sub>

**Comment by bonzani** [comment]
I have tested this item :white_check_mark: successfully on b3b9765a5622a89171daeedcba4af17797d574e2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30316">issues.joomla.org/tracker/joomla-cms/30316</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30316">issues.joomla.org/tracker/joomla-cms/30316</a>.</sub>

**Comment by wilsonge** [comment]
Thanks! (for this and the upstream PR!)

**Comment by nikosdion** [comment]
@wilsonge The upstream PR has been merged, see https://github.com/fgrosse/PHPASN1/pull/80#event-3681578783 You can run a `composer update` on Monday to pull the new code. If you also add a dependency to [phpseclib/bcmath_compat](https://github.com/phpseclib/bcmath_compat) the WebAuthn code will work even when neither the GMP nor the BCmath extensions are enabled on a server (I tested this by manually applying the upstream PR changes). This means that you can remove all of the now unnecessary checks for GMP in WebAuthn.

**Pull request merged**